meerk40t/core/cutcode.py

```diff
@@ -220,15 +221,15 @@ def contains_uncut_objects(self):
             return False
         for c in self.contains:
             for pp in c.flat():
-                if pp.permitted and pp.burns_remaining >= 1:
+                if pp.permitted and pp.burns_done < pp.passes:
```

Comment:
You don't need pp.permitted. The only thing it does is the same thing as `burns_done != 0` You can just remove it from everywhere as the other logic does the same thing.

Reply:
Yes - you have said this before and I have explained why I am reluctant to remove it (because it is used in `inner_selection_cutcode`) and have asked for an explanation as to what that does.

However, I have a hunch about what `inner_selection_cutcode` does, and I need to check the code to see if that is correct, and then see if I can find a better (and faster?) way to achieve the same goal without using `permitted`.

Reply:
I have now confirmed that this is OK to do and have made these changes, which I will push for review in a few minutes.

I will test these tomorrow evening.